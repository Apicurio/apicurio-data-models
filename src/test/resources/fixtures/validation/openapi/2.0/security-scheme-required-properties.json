{
    "swagger":"2.0",
    "info":{
        "description":"This is a sample server Petstore server via JSON.",
        "version":"1.0.0",
        "title":"Swagger Petstore (JSON)",
        "termsOfService":"http://helloreverb.com/terms/",
        "contact":{
            "email":"apiteam@wordnik.com"
        },
        "license":{
            "name":"Apache 2.0",
            "url":"http://www.apache.org/licenses/LICENSE-2.0.html"
        }
    },
    "host":"petstore.swagger.io",
    "basePath":"/v2",
    "tags":[
        {
            "name":"pet",
            "description":"Everything about your Pets",
            "externalDocs":{
                "description":"Find out more",
                "url":"http://swagger.io"
            }
        },
        {
            "name":"store",
            "description":"Operations about user"
        },
        {
            "name":"user",
            "description":"Access to Petstore orders",
            "externalDocs":{
                "description":"Find out more about our store",
                "url":"http://swagger.io"
            }
        }
    ],
    "schemes":[
        "http"
    ],
    "paths": {},
    "securityDefinitions":{
        "notype_auth":{
        },
        "apikey_auth_1":{
            "type": "apiKey"
        },
        "oauth2_auth_1":{
            "type": "oauth2"
        },
        "oauth2_auth_2":{
            "type": "oauth2",
            "flow": "implicit"
        },
        "oauth2_auth_3":{
            "type": "oauth2",
            "flow": "accessCode"
        },
        "oauth2_auth_4":{
            "type": "oauth2",
            "flow": "password"
        },
        "oauth2_auth_5":{
            "type": "oauth2",
            "flow": "application"
        }
    },
    "definitions":{
        "Order":{
            "properties":{
                "id":{
                    "type":"integer",
                    "format":"int64"
                },
                "petId":{
                    "type":"integer",
                    "format":"int64"
                },
                "quantity":{
                    "type":"integer",
                    "format":"int32"
                },
                "shipDate":{
                    "type":"string",
                    "format":"date-time"
                },
                "status":{
                    "type":"string",
                    "description":"Order Status",
                    "enum":[
                        "placed",
                        "approved",
                        "delivered"
                    ]
                },
                "complete":{
                    "type":"boolean"
                }
            },
            "xml":{
                "name":"Order"
            }
        },
        "Category":{
            "properties":{
                "id":{
                    "type":"integer",
                    "format":"int64"
                },
                "name":{
                    "type":"string"
                }
            },
            "xml":{
                "name":"Category"
            }
        },
        "User":{
            "properties":{
                "id":{
                    "type":"integer",
                    "format":"int64"
                },
                "username":{
                    "type":"string"
                },
                "firstName":{
                    "type":"string"
                },
                "lastName":{
                    "type":"string"
                },
                "email":{
                    "type":"string"
                },
                "password":{
                    "type":"string"
                },
                "phone":{
                    "type":"string"
                },
                "userStatus":{
                    "type":"integer",
                    "format":"int32",
                    "description":"User Status"
                }
            },
            "xml":{
                "name":"User"
            }
        },
        "Tag":{
            "properties":{
                "id":{
                    "type":"integer",
                    "format":"int64"
                },
                "name":{
                    "type":"string"
                }
            },
            "xml":{
                "name":"Tag"
            }
        },
        "Pet":{
            "required":[
                "name",
                "photoUrls"
            ],
            "properties":{
                "id":{
                    "type":"integer",
                    "format":"int64"
                },
                "category":{
                    "$ref":"#/definitions/Category"
                },
                "name":{
                    "type":"string",
                    "example":"doggie"
                },
                "photoUrls":{
                    "type":"array",
                    "xml":{
                        "name":"photoUrl",
                        "wrapped":true
                    },
                    "items":{
                        "type":"string"
                    }
                },
                "tags":{
                    "type":"array",
                    "xml":{
                        "name":"tag",
                        "wrapped":true
                    },
                    "items":{
                        "$ref":"#/definitions/Tag"
                    }
                },
                "status":{
                    "type":"string",
                    "description":"pet status in the store",
                    "enum":[
                        "available",
                        "pending",
                        "sold"
                    ]
                }
            },
            "xml":{
                "name":"Pet"
            }
        },
        "ApiResponse":{
            "properties":{
                "code":{
                    "type":"integer",
                    "format":"int32"
                },
                "type":{
                    "type":"string"
                },
                "message":{
                    "type":"string"
                }
            },
            "xml":{
                "name":"##default"
            }
        }
    },
    "externalDocs":{
        "description":"Find out more about Swagger",
        "url":"http://swagger.io"
    }
}