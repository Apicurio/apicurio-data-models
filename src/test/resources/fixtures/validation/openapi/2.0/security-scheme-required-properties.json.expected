[SS-001] |medium| {/securityDefinitions[notype_auth]->type} :: Security Scheme is missing a type.
[SS-002] |medium| {/securityDefinitions[apikey_auth_1]->name} :: API Key Security Scheme is missing a parameter name (e.g. name of a header or query param).
[SS-003] |medium| {/securityDefinitions[apikey_auth_1]->in} :: API Key Security Scheme must describe where the Key can be found (e.g. header, query param, etc).
[SS-004] |medium| {/securityDefinitions[oauth2_auth_1]->flow} :: OAuth Security Scheme is missing a flow type.
[SS-007] |medium| {/securityDefinitions[oauth2_auth_1]->scopes} :: OAuth Security Scheme is missing defined scopes.
[SS-005] |medium| {/securityDefinitions[oauth2_auth_2]->authorizationUrl} :: OAuth Security Scheme is missing an Authorization URL.
[SS-007] |medium| {/securityDefinitions[oauth2_auth_2]->scopes} :: OAuth Security Scheme is missing defined scopes.
[SS-005] |medium| {/securityDefinitions[oauth2_auth_3]->authorizationUrl} :: OAuth Security Scheme is missing an Authorization URL.
[SS-006] |medium| {/securityDefinitions[oauth2_auth_3]->tokenUrl} :: OAuth Security Scheme is missing a Token URL.
[SS-007] |medium| {/securityDefinitions[oauth2_auth_3]->scopes} :: OAuth Security Scheme is missing defined scopes.
[SS-006] |medium| {/securityDefinitions[oauth2_auth_4]->tokenUrl} :: OAuth Security Scheme is missing a Token URL.
[SS-007] |medium| {/securityDefinitions[oauth2_auth_4]->scopes} :: OAuth Security Scheme is missing defined scopes.
[SS-006] |medium| {/securityDefinitions[oauth2_auth_5]->tokenUrl} :: OAuth Security Scheme is missing a Token URL.
[SS-007] |medium| {/securityDefinitions[oauth2_auth_5]->scopes} :: OAuth Security Scheme is missing defined scopes.